### PR TITLE
Add dotnet-compat ABI testing and Debug configuration building to drone-CI

### PR DESCRIPTION
**Pull request by EraYaN**
**Changes**
Adds a whole bunch of stages to the drone-ci (but only for pull-requests)

### CODE REVIEW

**Comment by EraYaN** [comment]
@nvllsvm Could you kick off a drone-ci task again?

**Comment by JustAMan** [comment]
@EraYaN I restarted the task, it still doesn't run: https://cloud.drone.io/jellyfin/jellyfin/1489/1/11

File: .drone.yml [hunk 1]
```diff
@@ -6,7 +6,107 @@ steps:
...
+  - unzip -j Jellyfin.Naming.nupkg  "*.dll" -d /current-release
+  when:
+    event:
+    - pull_request
+
+- name: run-dotnet-compat-common
+  image: debian:stretch
+  err_ignore: true
+  commands:
+  - /tools/CompatibilityCheckerCoreCLI /current-release/Jellyfin.Common.dll /release/Jellyfin.Common.dll
```
Comment (EraYaN) [inline .drone.yml]: This is what the latest code looks like... Drone does not run that version.

**Comment by EraYaN** [comment]
This works now. Runs both correctly on the drone cloud and using `drone exec`

**Comment by EraYaN** [comment]
Updated the check tool to print some more information.

**Comment by EraYaN** [comment]
Right so the only solutions seems to be to talk to the GitHub statuses API directly or use for example Azure Pipelines to do the compat checks. (And later build our windows/macOS setups and what have you)

**Comment by EraYaN** [comment]
@jellyfin/core How to proceed?

**Comment by nvllsvm** [comment]
@EraYaN - I'd like CI to be all or nothing. If we're going Azure, make everything Azure. Having both would be too complicated.

**Comment by EraYaN** [comment]
Alright I do vote for Azure then for now, until drone is a bit more mature. Although it is not to outlandish to see projects use multiple services (since they do not want to pay). Not sure how portable the yaml is. I'll set it up in parallel first so we can test.

**Comment by nvllsvm** [comment]
You probably did this already, but have you checked with Drone support forums?

**Comment by EraYaN** [comment]
Yes I found the issue even that talks about this, it' not in the software yet and there is no implementation started.

**Comment by nvllsvm** [comment]
Please link to it :)

**Comment by EraYaN** [comment]
Alright so here is some of the issues: https://github.com/drone/drone/issues/1509

Here is some of why it's not possible yet/lots of work: https://discourse.drone.io/t/new-checks-api-by-github/2217/4

A prerequisite: 
https://github.com/drone/drone/issues/2422

And there are some more.
Like this would let you at least change the default context: https://github.com/drone/drone/issues/2577

There were some more, but I didn't bookmark those.

**Comment by nvllsvm** [comment]
Thank you!

**Pull request merged**